.github/workflows/test.yml

```
@@ -1,31 +1,31 @@
-name: main
+name: test
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+      - "test-me-*"
+  pull_request:
 
 # Cancel running jobs for the same workflow and branch.
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
 jobs:
-  build:
-
+  test:
     runs-on: ${{ matrix.os }}
 
     strategy:
       fail-fast: false
       matrix:
-        tox_env: ["py37", "py38", "py39", "py310"]
+        tox_env: ["py310", "py311"]
         os: [ubuntu-latest, windows-latest]
         include:
-          - tox_env: "py37"
-            python: "3.7"
-          - tox_env: "py38"
-            python: "3.8"
-          - tox_env: "py39"
-            python: "3.9"
           - tox_env: "py310"
             python: "3.10"
+          - tox_env: "py311"
+            python: "3.11"
```

This and the other suggestion about python version are exclusive.
```suggestion
          - tox_env: "py310"
            python: "3.10"
          - tox_env: "py311"
            python: "3.11"
          # Just checking syntax.
          - tox_env: "py38"
            python: "3.8"
            os: ubuntu-latest
          - tox_env: "py39"
            python: "3.9"
            os: ubuntu-latest
          
```